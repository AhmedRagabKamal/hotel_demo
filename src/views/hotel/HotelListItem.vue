<template>
  <div class="col-sm-4">
    <div class="card mb-3">
      <div class="card-body text-left">
        <button @click="getHotelDetail(hotel.id)"
         class="btn btn-link">{{hotel.name}}</button>
        <div class="row">
          <div class="col-sm-5">
              <img :src="hotel.photo" class="img-thumbnail" :alt="hotel.name" />
          </div>
          <div class="col-sm-7">
            <ul class="list-unstyled">
              <li>{{hotel.pricePerNight * nightsCount | currency('$')}}
                 for {{nightsCount}} night</li>
              <li>{{hotel.totalScore}}
                 <span v-grade="{grade:hotel.totalScore}"></span>
              </li>
              <li>{{hotel.totalReviews}} reviews</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import currency from '../../shared/filters/currency-filter';
import grade from '../../shared/directives/grade-directive';

export default {
  name: 'HotelListItem',
  props: {
    hotel: {
      required: true,
      type: Object,
      default: () => {},
    },
    nightsCount: {
      required: true,
      type: [Number, String],
      default: 1,
    },
  },
  methods: {
    getHotelDetail(hotelId) {
      this.$store.dispatch('hotel/getHotelDetails', hotelId);
    },
  },
  filters: {
    currency,
  },
  directives: {
    grade,
  },
};
</script>

<style scoped lang='scss'>

</style>
