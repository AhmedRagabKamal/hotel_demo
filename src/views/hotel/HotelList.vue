<template>
  <div>
    <div class="container mb-5">
      <div class="row">
        <hotel-list-item v-for="(hotel, index) in hotels"
         :key="index" :hotel="hotel" :nightsCount="nightsCount">
        </hotel-list-item>
      </div>
    </div>
  </div>
</template>

<script>
import HotelListItem from './HotelListItem.vue';
export default {
  name: 'HotelList',
  props: {
    nightsCount: {
      type: [Number, String],
      required: true,
      default: 1,
    },
  },
  created() {
    this.$store.dispatch('hotel/getHotels');
  },
  computed: {
    hotels() {
      return this.$store.state.hotel.hotels;
    }
  },
  components: {
    HotelListItem,
  },
}
</script>

<style>

</style>
