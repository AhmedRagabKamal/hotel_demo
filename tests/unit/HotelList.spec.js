import { shallowMount } from '@vue/test-utils';
import HotelList from '@/views/hotel/HotelList.vue';
import HotelListItem from '@/views/hotel/HotelListItem.vue';

describe('HotelList', () => {
  const build = () => {
    const wrapper = shallowMount(HotelList, {
      propsData: {
        nightsCount: 1,
      },
    });
    return {
      wrapper,
      hotelListItem: () => wrapper.find(HotelListItem),
    };
  };

  it('HotelList render ok', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main child component HotelListItem', () => {
    // arrange
    const { hotelListItem } = build();
    // assert
    expect(hotelListItem().exists()).toBe(true);
  });

  // it('passes a binded nightCounts prop to HotelListItem component', () => {
  //   // arrange
  //   const days = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  //   const { wrapper, hotelListItem } = build();
  //   wrapper.setData({
  //     nightsCount: 1,
  //   });
  //   // assert
  //   expect(hotelListItem().vm.nightsCount).toBe(wrapper.vm.nightsCount);
  //   expect(wrapper.vm.nightsCount).not.toBeUndefined();
  //   expect(wrapper.vm.nightsCount).toBeDefined();
  //   expect(wrapper.vm.nightsCount).toBeTruthy();
  //   expect(days).toContain(wrapper.vm.nightsCount);
  // });


});
