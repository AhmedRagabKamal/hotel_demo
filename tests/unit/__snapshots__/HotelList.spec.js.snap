// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList HotelList render ok 1`] = `
<div>
  <div class="container mb-5">
    <div class="row"></div>
  </div>
</div>
`;
